lib/ddtrace/contrib/active_record/utils.rb

```diff
@@ -23,14 +25,9 @@ def self.connection_config(connection = nil)
           connection.nil? ? default_connection_config : connection_config_from_connection(connection)
         end
 
-        # Attempt to retrieve the connection config from an object ID.
         # Typical of ActiveSupport::Notifications `sql.active_record`
         def self.connection_config_from_connection(connection)
-          if connection.instance_variable_defined?(:@config)
```

Comment:
I think we need to keep the `defined?` check here, otherwise this generates warnings.